### PR TITLE
@craigspaeth : replace google search with elasticsearch

### PR DESCRIPTION
**Pull request by cavvia**
* Switches our 'search results' page over from Google to Elasticsearch. 
* Closely mirrors the corresponding artsy/force#410 PR.
* Uses our existing `SearchResult` model and collection, deleting the `GoogleSearchResult` equivalents. 
* Adds description formatting logic for shows/fair booths. Previously this was returned by the Google API but microgravity now needs to construct it.

I've also tested this functionally locally and it looks fine in terms of result rendering.

### CODE REVIEW

File: models/fair_location.coffee [hunk 1]
```diff
@@ -2,10 +2,9 @@ _ = require 'underscore'
 sd = require('sharify').data
 Backbone = require 'backbone'
 Relations = require './mixins/relations/location.coffee'
+Location = require './location.coffee'
 
-module.exports = class FairLocation extends Backbone.Model
-
-  _.extend @prototype, Relations
+module.exports = class FairLocation extends Location
```
Comment (cavvia) [inline models/fair_location.coffee:7]: Added inheritance chain here in order to reuse the `singleLine` formatting method. This echoes our Force data model. I also do this with `PartnerLocation` below.

File: test/models/search_result.coffee [hunk 2]
```diff
@@ -31,22 +36,128 @@ describe 'SearchResult', ->
...
+
+      result = new SearchResult(profile)
+      result.about().should.equal 'A description of foo gallery'
+
+  describe '#formatCityAbout', ->
+    it 'uses a standard template', ->
+      result = new SearchResult({ model: 'city', display: 'Gotham', id: 'gotham' })
+      result.about().should.equal 'Browse current exhibitions in Gotham'
+
+  describe '#status', ->
```
Comment (craigspaeth) [inline test/models/search_result.coffee:134]: Nice test coverage + use of model-layer logic. 👍

File: models/search_result.coffee [hunk 3]
```diff
@@ -74,3 +93,70 @@ module.exports = class SearchResult extends Backbone.Model
       @set display_model: 'Booth'
     else
       @set location: "#{fair.href()}/browse#{@get('location')}"
+
+  status: ->
+    if @get('model') == 'partnershow'
+      if startTime = @get('start_at')
+        if endTime = @get('end_at')
```
Comment (craigspaeth) [inline models/search_result.coffee]: These nested `if`s could maybe be easier to follow flattened, e.g.

````coffeescript
return unless @get('model') == 'partnershow'
startTime = moment @get('start_at')
endTime = moment @get('end_at')

if moment().isAfter(endTime)
// ...
````

File: models/search_result.coffee [hunk 4]
```diff
@@ -74,3 +93,70 @@ module.exports = class SearchResult extends Backbone.Model
...
+          if moment() > moment(endTime)
```
Comment (craigspaeth) [inline models/search_result.coffee]: I'm surprised this works, maybe what you're looking for is [moment().isAfter](https://momentjs.com/docs/#/query/is-after/).

**Comment by craigspaeth** [comment]
Looks like some related route spec failures, otherwise no blocking comments on my end. Feel free to address and merge when 💚  Thanks! Solid work!

```
  1) Search routes #index success makes the appropriate request and removes accents:
     TypeError: Cannot read property 'should' of undefined
    at Context.<anonymous> (apps/search/test/routes.coffee:25:40)
  

  2) Search routes #index error doesnt swallow the error:
     Uncaught AssertionError: expected false to be true
    at Assertion.fail (node_modules/should/cjs/should.js:205:17)
    at Assertion.value (node_modules/should/cjs/should.js:277:19)
    at apps/search/test/routes.coffee:55:11
    at Timeout._onTimeout (node_modules/underscore/underscore.js:768:19)
```

File: models/search_result.coffee (hunk 3 above)
Comment (cavvia) [reply models/search_result.coffee]: sure, i'll refactor this.

**Comment by cavvia** [comment]
thanks @craigspaeth - i had difficulty testing the error callback in the route (as i had changed the scheme from a `.fetch.then.catch` promise based scheme to a `.fetch` with success and error callback params). so i had to delete that spec in the end. any advice on how best to test that would be appreciated - i think i may need to use a `sinon.fakeServer`.

**Pull request merged**